src/Tests/ImageBundleTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\media\Tests\ImageBundleTest.
+ */
+
+namespace Drupal\media\Tests;
+
+use Drupal\file\Entity\File;
+use Drupal\simpletest\WebTestBase;
+
+/**
+ * Ensures that media bundle for images can be created.
+ *
+ * @group media
+ */
+class ImageBundleTest extends WebTestBase {
+
+  /**
+   * Modules to enable.
+   *
+   * @var array
+   */
+  public static $modules = ['media_entity', 'media_entity_image', 'image', 'node'];
+
+  /**
+   * The test media bundle.
+   *
+   * @var \Drupal\media_entity\MediaBundleInterface
+   */
+  protected $testBundle;
+  /**
+   * Contains the media bundle storage
+   *
+   * @var \Drupal\media_entity\MediaBundleInterface
+   */
+  protected $bundleStorage;
+
+  /**
+   * The test user.
+   *
+   * @var \Drupal\User\UserInterface
+   */
+  protected $adminUser;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->container->get('module_installer')->install(['image_bundle_test']);
+    $entityManager = $this->container->get('entity.manager');
+    $this->bundleStorage = $entityManager->getStorage('media_bundle');
+    $this->testBundle = $this->bundleStorage->load('images_media_bundle');
+
+    $this->adminUser = $this->drupalCreateUser([
+        'view media',
+        'create media',
+        'update media',
+        'update any media',
+        'delete media',
+        'delete any media',
+        'access media overview',
+    ]);
+    $this->drupalLogin($this->adminUser);
+  }
+
+  /**
+   * Tests image media bundle creation from config files.
+   */
+  public function testMediaBundleCreationFromModule() {
+    $type_configuration = array(
+      'source_field' => 'field_image',
+      'gather_exif' => false
+    );
+
+    $this->assertTrue((bool) $this->testBundle, 'The media bundle from default configuration has been created in the database.');
+    $this->assertEqual($this->testBundle->get('label'), 'Images media bundle', 'Correct label detected.');
+    $this->assertEqual($this->testBundle->get('description'), 'Media bundle for Images', 'Correct description detected.');
+    $this->assertEqual($this->testBundle->get('type'), 'image', 'Correct plugin ID detected.');
+    $this->assertEqual($this->testBundle->get('type_configuration'), $type_configuration, 'Type configuration correct.');
+    $this->assertEqual($this->testBundle->get('field_map'), [], 'Correct field map detected.');
+  }
+
+  /**
+   * Tests item creation and thumbnail
+   */
+  public function testMediaBundleItemCreation(){
+    // Define the media item name.
+    $name = $this->randomMachineName();
+
+    $image_files = $this->drupalGetTestFiles('image');
+    $testImage = File::create((array) current($image_files));
```

Comment:
I don't quite understand this. uploadNodeImage posts to node/add/$type but we need to post to /media/add/$type to test. Also the method requires a file object $image which will have to be generated using:

`$image_files = $this->drupalGetTestFiles('image');
$testImage = File::create((array) current($image_files));
`